cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ResourceOperatorSupplier.java

```diff
@@ -302,7 +304,8 @@ public ResourceOperatorSupplier(Vertx vertx,
                                     MetricsProvider metricsProvider,
                                     ZooKeeperAdminProvider zkAdminProvider,
                                     PlatformFeaturesAvailability pfa,
-                                    long operationTimeoutMs) {
+                                    long operationTimeoutMs,
+                                    BrokersInUseCheck brokersInUseCheck) {
```

Comment:
Why was this needed? Also, if it is really needed, you should probably tr to maintain some ordering and move it at least above the timeout.

Reply:
I think we had this discussion offline, but this is to be able to mock the `BrokersInUseCheck` class in the tests where I need to specify nodes which are blocked on scale down. So I can pass the mock to the supplier and from the supplier I can get the mock and mocking like this:

```java
BrokersInUseCheck operations = supplier.brokersInUseCheck;
when(operations.brokersInUse(any(), any(), any(), any())).thenReturn(Future.succeededFuture(Set.of(3, 4)));
```

You can find this usage in the `KafkaAutoRebalancingMockTest` class.

Reply:
Anyway I will move the parameter up in the list.

Reply:
I do not remember such discussion. You can easily mock this by mocking the Kafka Admin API without adding new parameters and introducing them to many different classes.

Reply:
That was another option but tbh this one looked easier to me.

Reply:
I will give the admin client mocking a try.

Reply:
You can pretty uch copy the mock from BrokersInUseCheckTest and just wrap it into a method to return exactly the node IDs in use you need.

Reply:
I have switched to mock admin client as suggested.